community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -390,6 +392,9 @@ boolean applicable( DiagnosticsPhase phase )
     private StoreLayerModule storeLayerModule;
     private TransactionLogModule transactionLogModule;
     private KernelModule kernelModule;
+    private BufferingIdGeneratorFactory bufferingIdGeneratorFactory;
+
+    private final Predicate<KernelTransactionsSnapshot> eligibleForReuse;
```

Comment:
According to usages of this predicate, argument `KernelTransactionsSnapshot` is never used.
I propose to create a special type for this check like:

```
interface IdReuseCheck {
    boolean safeToReuse();
}
```

and use it instead of the predicate. It'll make it clearer what is this check about and easier to find usages of `safeToReuse()`.